### ISSUE TITLE
Add /etc/sysconfig/proxy to the network environment

### ISSUE THREAD

**Issue by mbukva1** [SUSE/suse-migration-services#180]
Some customers, who for whatever reason are unable to use transparent proxies in their networks, still use /etc/sysconfig/proxy to define system wide proxies. I understand the Distribution Migration System does not carry this file across, so the migration always fails because the migration system cannot find scc.suse.com without the proxy defined. 

Note that my suggestion is only a solution for what I expect to be the majority of cases - where customers set system-wide proxies in /etc/sysconfig/proxy AND do not require a password for the proxy. Some customers use environment variables rather than /etc/sysconfig/proxy and some do also require passwords for their proxies, for example stored in .curlrc.

**Comment by schaefi** [comment]
We already have a preserve feature in place. Currently only files from the udev rules area can be preserved but this feature can be extended such that you can make use of it.

**Comment by mbukva1** [comment]
Hi Marcus,

Are you suggesting that myself or the customer fork the project and make the change in whichever bit of python code does the preserving of files ? For myself, I’m not a python developer at all, and neither to my knowledge are the customer’s linux/SAP Basis admins. Apologies for any misunderstanding of the process you had in mind here.

Regards,
-Michael Bukva

**Comment by rjschwei** [comment]
> Hi Marcus, Are you suggesting that myself or the customer fork the project and make the change in whichever bit of python code does the preserving of files ? 

No that is not the suggestion. @jesusbv will pick this up and develop the necessary code changes.

**Comment by mbukva1** [comment]
Hi Robert,

I really appreciate the clarification. Is it possible to project a timeframe for a customer-accessible update? I’m looking to give my customer’s project team some guidance, if this work is happening ‘soon’ that would be great.

In terms of a “plan B” for their 12->15 upgrades they have a SUMA installed in the public cloud for their patching but no in house capability of upgrades via the… complexity… of autoyast profiles. Hence the strong interest in DMS!

Regards,
-Michael Bukva

**Comment by schaefi** [comment]
@mbukva1 Hi. As Robert said, we are on the hook to come up with an implementation. 

With regards to the timeframe I will come back to you today with an answer. I'd like to clarify with @jesusbv  first.

> Hence the strong interest in DMS!

Happy to hear that :)

**Comment by schaefi** [comment]
> Is it possible to project a timeframe for a customer-accessible update?

ok here is the plan. Implementation/testing and the submission to the SUSE namespace will be done by the end of this week.
You should plan for another week for maintenance and QA to approve our submission. Does this help to coordinate with the customer ?

Thanks

**Comment by mbukva1** [comment]
Yes it does - thank you!

Regards,
-Michael Bukva

Sent from my iPhone

On 1 Feb 2021, at 20:58, Marcus Schäfer <notifications@github.com> wrote:

﻿

Is it possible to project a timeframe for a customer-accessible update?

ok here is the plan. Implementation/testing and the submission to the SUSE namespace will be done by the end of this week.
You should plan for another week for maintenance and QA to approve our submission. Does this help to coordinate with the customer ?

Thanks

**Comment by schaefi** [comment]
A package submission has been done

```
Request: #235410

  maintenance_incident: home:sax2:branches:SUSE:SLE-15:Update/suse-migration-services@b237da6ea31fc4ecd7544e879ba4f558 -> SUSE:Maintenance (release in SUSE:SLE-15:Update)


Message:
- Bump version: 2.0.15 → 2.0.16
- 
- Extend preserved files
  Add option to preserve files.
  This fixes #180
  This references bsc#1181808
```

we now have to wait for the QA team to approve this. Once done we will submit a new image with this package.
After that submission and approval customers can use it

**Comment by jirib** [comment]
Hi, a have a support case where customer was using proxy, thus IMO this should be preserved by default.
